src/versionbits.h

```diff
@@ -70,21 +72,32 @@ class AbstractThresholdConditionChecker {
     int GetStateSinceHeightFor(const CBlockIndex* pindexPrev, const Consensus::Params& params, ThresholdConditionCache& cache) const;
 };
 
-/** BIP 9 allows multiple softforks to be deployed in parallel. We cache per-period state for every one of them
- *  keyed by the bit position used to signal support. */
-struct VersionBitsCache
+/** BIP 9 allows multiple softforks to be deployed in parallel. We cache
+ *  per-period state for every one of them. */
+
+class VersionBitsCache
```

Comment:
I don't think I agree -- if there wasn't a cache, there would be no need for an object, these things would just be standalone functions, so calling the object a cache makes sense to me.

There's some impedence mismatch though -- apart from having to have the object, the api is otherwise pretty independent of whether there's a cache or not.

Reply:
> apart from having to have the object, the api is otherwise pretty independent of whether there's a cache or not.

Indeed. The client doesn't care whether there's a cache or not (and for the static functions, there isn't even a cache). Generally, I think interfaces should be named according to the client usage, not the implementation specifics. Not a big deal, and shouldn't hold up this PR.

Reply:
The client does care a little about whether there's a cache -- without it, checking the state is potentially linear in the height of the block being tested, with it, it's amortized constant. Particularly for deployments with short periods (like bip91) that could be prohibitive.

Perhaps worth consider renaming it to `SignalledDeploymentManager` or similar along with renaming from `BIP9` to `VBits` or renaming `DeploymentPos` or a deeper refactoring of `AbstractThresholdConditionChecker` etc.

Reply:
I'd prefer a generic name like `SignalledDeploymentManager`, but it definitely shouldn't hold up this PR.

Reply:
> The client does care a little about whether there's a cache

Sorry, I meant in terms of correctness/observability. Of course a cache has implications for computational complexity.